rebalance/rebalance.py

```diff
@@ -620,16 +633,68 @@ def maybe_rebalance_pairs(ch1, ch2, failed_channels: list):
     return result
 
 
+def rebalance_pair_picker(threadid, channel_pairs: list, failed_channels: list):
+    plugin.threadids[threading.get_ident()] = threadid
+    result = {"success": False, "fee_spent": Millisatoshi(0)}
+    idle_count = 0
```

Comment:
Can you explain (to me) the usage of this `idle_count` variable?
It looks to me like it is increased whenever an unfitting or already processed pair is found. Its then set to `0` when it will actually do something. It will log a message when it was `1`, and wait 10 secs if its greater than `0` in the end. It is never decreased (Other than explicitly setting it to `0` once) . Is this correct?

Maybe my understanding of Threads in Python is not reflecting reality :D, but isn't `idle_count` (and all the other variables in this function, like also `result`) on a per-thread basis so that any thread has its own distinct version of this variable?

Reply:
`idle_count`: it is on a per-thread basis, so every thread has its own `idle_count`.
If the thread can try at least one rebalance, the `idle_count` become zero. If the thread checked all possible pairs, and cannot do anything, then `idle_count` increases. Means that this thread is idle. So the thread goes to sleep (as nothing to do), and logs that it is idle (but only once, when the `idle_count` equals 1, to not to spam the log).

The idea on the background: it is difficult to get the ideal number of threads. My node has 40 active channels, and in the beginning of a `rebalanceall` up to 12 threads can work parallel. After a time, some threads cannot find a job. Nearing the end, it's only 1 or 2 threads are working, because of the few possibilities. But this is not a continuous decrease in the number of working thread, because sometimes as a rebalance attempt finishes, two others can start, due to the channels being released. So it is not a good concept to terminate the idling thread, as they can be useful later.